app/app-controllers/segmentedController.js

```diff
@@ -0,0 +1,78 @@
+define([
+    'bluebird',
+    'plugins/anchoredLayoutPlugin',
+    'plugins/segmentedPlugin',
+    'config/baseConfig',
+    'config/segmentsConfig'
+],
+function(
+    Promise,
+    AnchoredLayoutPlugin,
+    SegmentedPlugin,
+    BaseConfig,
+    SegmentsConfig
+) {
+    var SegmentedController = function(layout, navigationView, config, urlMap) {
+        this.parentLayout = layout;
+        this.navigationView = navigationView;
+        this.config = config;
+        this.urlMap = urlMap;
+        this.currentSegments = null;
+        this.registerEvents();
+    };
+
+    SegmentedController.init = function(layoutPromise, navigationViewPromise) {
+        var urlMap = {};
+        return Promise.join(
+            layoutPromise,
+            navigationViewPromise,
+            function(layout, navigationView) {
+                return Promise.all(
+                    SegmentsConfig.map(function(page, _) {
+                        return SegmentedPlugin.init().then(function(segmentedControls) {
```

Comment:
It feels weird having the `segmentedControls` parameter be plural... it's just a single object, isn't it?